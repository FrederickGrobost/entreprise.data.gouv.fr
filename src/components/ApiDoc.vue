<template>
  <div>
    <api-doc-intro />
    <api-doc-list />
    <api-doc-others />
    <api-doc-outro />
  </div>
</template>

<script>
import ApiDocIntro from "@/components/apiDoc/ApiDocIntro";
import ApiDocList from "@/components/apiDoc/ApiDocList";
import ApiDocOthers from "@/components/apiDoc/ApiDocOthers";
import ApiDocOutro from "@/components/apiDoc/ApiDocOutro";

export default {
  name: "ApiDoc",
  metaInfo: {
    title: "Documentation"
  },
  components: {
    ApiDocIntro: ApiDocIntro,
    ApiDocList: ApiDocList,
    ApiDocOthers: ApiDocOthers,
    ApiDocOutro: ApiDocOutro
  }
};
</script>
