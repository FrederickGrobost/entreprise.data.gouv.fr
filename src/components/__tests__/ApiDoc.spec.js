import { createLocalVue, mount, RouterLinkStub } from "@vue/test-utils";
import { __createMocks as createStoreMocks } from "@/store/index.js";
import Vuex from "vuex";
import ApiDoc from "@/components/ApiDoc";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.component("router-link", RouterLinkStub);
jest.mock("@/store/index.js");

describe("ApiDoc.vue snapshot testing", () => {
  const storeMocks = createStoreMocks();
  const wrapperAD = mount(ApiDoc, {
    localVue,
    store: storeMocks.store
  });

  test("It match the snapshot", () => {
    expect(wrapperAD.vm.$el).toMatchSnapshot();
  });
});
