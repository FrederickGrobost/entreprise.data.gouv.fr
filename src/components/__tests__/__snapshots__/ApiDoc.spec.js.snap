// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiDoc.vue snapshot testing It match the snapshot 1`] = `
<div>
  <section
    class="section section-white"
  >
    <div
      class="container text-center"
    >
      <h1>
        Les APIs Open Data
      </h1>
       
      <p
        class="text-center lead-text"
      >
        
      Entreprise.data.gouv.fr met à votre disposition des APIs open-source gratuites
        <br />
        
      pour faciliter l'utilisation des données par les professionnels et les particuliers.
        <br />
      </p>
       
      <p
        class="text-center lead-text"
      >
        
      De nombreuses APIs sont en cours d’ajout.
      Vous pouvez vous inscrire à notre mailing-list pour vous tenir au courant
      des nouveautés et changements à venir.
        <br />
      </p>
       
      <form
        class="text-center"
      >
        <h4>
          Inscription Mailing-List
        </h4>
         
        <p>
          
    Vos données sont traitées par Mailjet.
          <br />
           Vos informations ne seront jamais utilisées à des fins commerciales.
  
        </p>
         
        <iframe
          frameborder="0"
          height="95"
          marginheight="0"
          marginwidth="0"
          scrolling="no"
          src="https://app.mailjet.com/widget/iframe/zR1/ack"
          width="340"
        />
      </form>
    </div>
  </section>
   
  <div>
    <section
      class="section section-grey"
    >
      <div
        class="container apis"
      >
        <h2>
          L’API Sirene
        </h2>
         
        <div
          class="row"
        >
          <p
            class="description"
          >
            
          L’API Sirene contient tous les établissements des entreprises françaises,
          ainsi que toutes les associations disposant d’un code SIRET.
          Les données de l’API Sirene sont délivrées par l’INSEE.
        
          </p>
           
          <div
            class="cta"
          >
            <a
              class="button"
            >
              Documentation API Sirene
            </a>
          </div>
        </div>
      </div>
    </section>
     
    <section
      class="section section-white"
    >
      <div
        class="container apis"
      >
        <h2>
          L’API Repertoire National des Associations (RNA)
        </h2>
         
        <div
          class="row"
        >
          <p
            class="description"
          >
            
          Cette API contient toutes les associations françaises.
          Les données proviennent du Ministère de l’Intérieur.
        
          </p>
           
          <div
            class="cta"
          >
            <a
              class="button"
            >
              Documentation API RNA
            </a>
          </div>
        </div>
      </div>
    </section>
     
    <section
      class="section section-grey"
    >
      <div
        class="container apis"
      >
        <h2>
          L’API Répertoire National des Métiers (RNM)
        </h2>
         
        <div
          class="row"
        >
          <p
            class="description"
          >
            
          L’API RNM vous est gracieusement ouverte par les Chambres et Métiers de l’Artisanat.
          Les données sont mises à jour toutes les nuits.
        
          </p>
           
          <div
            class="cta"
          >
            <a
              class="button"
              href="https://api-rnm.artisanat.fr/"
            >
              Documentation API RNM
            </a>
          </div>
        </div>
      </div>
    </section>
     
    <section
      class="section section-white"
    >
      <div
        class="container apis"
      >
        <h2>
          L’API Registre National du Commerce et des Sociétés (RNCS)
        </h2>
         
        <div
          class="row"
        >
          <p
            class="description"
          >
            <strong>
              ⚠ En version alpha ⚠ 
            </strong>
            <br />
            
          Cette API contient les données IMR (Immatriculation, Modification, Radiation) du Registre National du Commerce et des Sociétés de l’INPI.
        
          </p>
           
          <div
            class="cta"
          >
            <a
              class="button"
            >
              Documentation API RNCS
            </a>
          </div>
        </div>
      </div>
    </section>
  </div>
   
  <section
    class="section section-grey"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="column__right"
        >
          <h2>
            Découvrez les autres APIs et jeux de données Data.gouv
          </h2>
           
          <p>
            <a
              href="https://data.gouv.fr"
            >
              Data.gouv.fr
            </a>
             met a votre disposition de nombreux jeux de données "Open Data".
            <br />
            
          Quelques exemples :
        
          </p>
           
          <ul>
            <li>
              <a
                href="https://www.data.gouv.fr/fr/datasets/resultats-de-lintegralite-des-elections-depuis-2001/"
              >
                Les résultats électoraux
              </a>
            </li>
             
            <li>
              <a
                href="https://www.data.gouv.fr/fr/datasets/fichier-des-prenoms-edition-2016/"
              >
                Le fichier des prénoms
              </a>
            </li>
             
            <li>
              <a
                href="https://www.data.gouv.fr/fr/datasets/secteurs-de-localisation-de-la-loutre-dans-le-jura/"
              >
                Les secteurs de localisation de la loutre dans le Jura
              </a>
            </li>
          </ul>
           
          <p>
            
          Vous pouvez également consulter des réutilisations utiles de ces jeux de données.
        
          </p>
           
          <h3>
            La mission Etalab
          </h3>
           
          <p>
            
          Etalab coordonne la politique d’ouverture et de partage des données publiques (“open data”). 
          
            <br />
             
            <a
              href="https://www.etalab.gouv.fr/en/qui-sommes-nous"
            >
              Plus d’informations.
            </a>
          </p>
        </div>
         
        <div
          class="column__icon"
        >
          <img
            alt=""
            src="@/assets/img/icons/statistics-on-a-screen.svg"
          />
        </div>
      </div>
    </div>
  </section>
   
  <section
    class="section section-white"
  >
    <div
      class="container text-center"
    >
      <h2>
        Notre autre produit : l'API Entreprise.
      </h2>
       
      <div
        class="column__img"
      >
        <img
          alt=""
          src="@/assets/img/city.svg"
        />
      </div>
       
      <h3>
        Vous justifiez d’une mission de service public ? L’API Entreprise est pour vous.
      </h3>
       
      <p>
        
      Dans le respect du principe "Dites le nous une seule fois", la DINSIC a mis en place l’API Entreprise,
      l’API qui aggrège les informations sur les entreprises.
        <br />
        
      Obtenez les Greffes, Certificats, Liasses fiscales, Attestations sociales, et bien plus encore...
        <br />
      </p>
       
      <p>
        <strong>
          Attention :
        </strong>
         Ces données ne sont pas publiques et sont donc reservées
      aux administrations ou entreprises justifiant d'une mission de service public.
        <br />
        
      Pour toutes questions : 
        <a
          href="maito:tech@entreprise.api.gouv.fr"
        >
          tech@entreprise.api.gouv.fr
        </a>
      </p>
       
      <a
        class="button"
        href="https://entreprise.api.gouv.fr/"
      >
        Demander un accès
      </a>
    </div>
  </section>
</div>
`;
